src/stochastic_triple.jl

```diff
@@ -94,10 +94,11 @@ end
 
 ### Extra overloads
 
-# TODO: generalize the below logic to compactly handle a wider range of functions
+# TODO: generalize the below logic to compactly handle a wider range of functions.
 # See also https://github.com/JuliaDiff/ForwardDiff.jl/blob/master/src/dual.jl.
 
 Base.hash(st::StochasticAD.StochasticTriple) = hash(StochasticAD.value(st))
+Base.hash(st::StochasticAD.StochasticTriple, hsh::UInt) = hash(StochasticAD.value(st), hsh)
```

Comment:
We could document somewhere that stochastic triples hash-compare equal with their values

Reply:
This would be a dangerous thing to do when the stochastic triple has a perturbation, so I've made the rule a bt safer (errors when there exists a perturbation), with the intention of generalizing and improving these ad-hoc rules later. As for the behaviour when there are no perturbations, it's a little tricky and probably deserves some more thought -- I just copied `ForwardDiff.jl` for now.